crates/sui-framework-build/src/compiled_package.rs

```diff
@@ -303,10 +303,10 @@ impl CompiledPackage {
         ids.into_iter().collect()
     }
 
-    pub fn get_package_digest(&self) -> [u8; 32] {
+    pub fn get_package_digest(&self, with_unpublished_deps: bool) -> [u8; 32] {
         MovePackage::compute_digest_for_modules_and_deps(
-            &self.get_package_bytes(/* with_unpublished_deps */ true),
-            &self.get_dependency_original_package_ids(),
+            &self.get_package_bytes(with_unpublished_deps),
+            self.dependency_ids.published.values(),
```

Comment:
cc @rvantonder -- when you add the support for checking that `unpublished` dependencies have `0x0` addresses, you will probably need to update this codepath as well to return a `Result`.